asapdiscovery-ml/asapdiscovery/ml/utils.py

```diff
@@ -1722,9 +1745,15 @@ def update_loss_dict(split, compound_id, target, in_range, uncertainty, pred, lo
 
         # Stop if loss has gone to infinity or is NaN
         if (
-            np.isnan(epoch_val_loss)
+            np.isnan(epoch_train_loss)
```

Comment:
Move to a function called `check_valid` or something like that this is a few too many `or`.